node/src/tasks/bsp_upload_file.rs

```diff
@@ -252,6 +252,33 @@ where
             .map_err(|_| anyhow::anyhow!("File key and HasherOutT mismatch!"))?;
         self.file_key_cleanup = Some(file_key_hash);
 
+        // Get the node's public key needed for threshold calculation.
+        let node_public_key = self
+            .storage_hub_handler
+            .blockchain
+            .get_node_public_key()
+            .await;
+
+        // Query runtime for the earliest block where the BSP can volunteer for the file.
+        let earliest_volunteer_block = self
+            .storage_hub_handler
+            .blockchain
+            .query_file_earliest_volunteer_block(
+                node_public_key,
+                H256::from_slice(file_key.as_ref()),
+            )
+            .await
+            .map_err(|e| anyhow!("Failed to query file earliest volunteer block: {:?}", e))?;
+
+        // If earliest_volunteer_block is 0 it means that we can volunteer right away.
```

Comment:
I would change this behaviour so that if the node can volunteer right away, it returns the current block instead of 0. Otherwise, you could make it so that it returns how many blocks you'd have to wait, then 0 would mean now, but that would be a bigger change.